src/foundry/common/abstract/socket.d.mts

```diff
@@ -0,0 +1,36 @@
+import type { DocumentType } from "../../../types/helperTypes.d.mts";
+import type { DatabaseAction, DatabaseOperation, DocumentSocketRequest } from "./_types.d.mts";
+
+/**
+ * The data structure of a modifyDocument socket response.
+ */
+declare class DocumentSocketResponse<Action extends DatabaseAction> {
+  /**
+   * Prepare a response for an incoming request.
+   * @param request - The incoming request that is being responded to
+   */
+  constructor(request: DocumentSocketRequest<Action>);
+
+  /** The type of Document being transacted. */
+  type: DocumentType | undefined;
+
+  /** The database action that was performed. */
+  action: Action | undefined;
+
+  /** Was this response broadcast to other connected clients? */
+  broadcast: boolean | undefined;
+
+  /** The database operation that was requested. */
+  operation: DatabaseOperation | undefined;
+
+  /** The identifier of the requesting user. */
+  userId: string | undefined;
+
+  /** The result of the request. Present if successful */
+  result: Record<string, unknown>[] | string[] | undefined;
+
+  /** An error that occurred. Present if unsuccessful */
+  error: Error | undefined;
```

Comment:
I looked in Foundry source and this mostly seems to be backend related. I was going to suggest that you identify if there's any items that are always sent, i.e. always required, but honestly this seems like a lot of effort for such a niche type and not necessarily forward compatible anyways.